src/graph/planner/ngql/GoPlanner.cpp

```diff
@@ -626,18 +640,31 @@ StatusOr<SubPlan> GoPlanner::transform(AstContext* astCtx) {
 }
 
 bool GoPlanner::isSimpleCase() {
-  if (goCtx_->joinDst || goCtx_->filter || !goCtx_->distinct || !goCtx_->limits.empty()) {
+  if (!goCtx_->onlyYieldDistinctDstId) {
+    return false;
+  }
+  if (goCtx_->joinDst || goCtx_->filter || !goCtx_->limits.empty()) {
     return false;
   }
   auto& exprProps = goCtx_->exprProps;
   if (!exprProps.srcTagProps().empty()) return false;
   if (!exprProps.dstTagProps().empty()) return false;
-  for (auto& edgeProp : exprProps.edgeProps()) {
+
+  return goCtx_->edgePropsOnlyUseDstId;
+}
+
+bool GoPlanner::edgePropsOnlyUseDstId() {
+  for (auto& edgeProp : goCtx_->exprProps.edgeProps()) {
     auto props = edgeProp.second;
     if (props.size() != 1) return false;
     if (props.find(kDst) == props.end()) return false;
   }
 
+  return true;
+}
+
+bool GoPlanner::onlyYieldDistinctDstId() {
+  if (!goCtx_->distinct) return false;
   if (goCtx_->yieldExpr->columns().size() != 1) {
     return false;
```

Comment:
Foreach yield items and check if expression contains only constant or id expressions.

Reply:
yield distinct like._dst, like._dst,8,7